Lib/unittest/test/testmock/testasync.py

```diff
@@ -0,0 +1,433 @@
+import asyncio
+import inspect
+import unittest
+
+from unittest.mock import call, AsyncMock, patch, MagicMock
+
+class AsyncClass:
+    def __init__(self):
+        pass
+    async def async_method(self):
+        pass
+    def normal_method(self):
+        pass
+
+async def async_func():
+    pass
+
+def normal_func():
+    pass
+
+class NormalClass(object):
+    def a(self):
+        pass
+
+
+async_foo_name = f'{__name__}.AsyncClass'
+normal_foo_name = f'{__name__}.NormalClass'
+
+
+class AsyncPatchDecoratorTest(unittest.TestCase):
+    def setUp(self):
+        # Prevents altering the execution environment
+        self.old_policy = asyncio.events._event_loop_policy
```

Comment:
Is this required?
AsyncMock doesn't change the asyncio loop/policy state anymore

Reply:
So if I run the tests without this, I get an error that I am altering the execution environment:
```
1 test altered the execution environment:
    test_unittest
```
I thought it was happening in _AwaitEvent? But I am not sure.

Reply:
Maybe that's because `asyncio.run()` alters the environment by definition.
I'll take a look.

Reply:
I figured out the issue.
Before any asyncio calls, the default event loop policy is `None`.
To instantiate a loop you need a policy.
The policy is created on demand.
So, basically after any loop creation, the policy is not `None` but something concrete like `_UnixDefaultEventLoopPolicy`.

Cleaning it up at the end of test is correct.
To simplify the code you can add simple module teardown function: 
```
def tearDownModule():
    asyncio.set_event_loop_policy(None)
```
asyncio tests do this, I've missed that fact.

Reply:
Thanks for digging into this @asvetlov! That makes sense. I'll update to use tearDownModule().